src/lib/types.hpp

```diff
@@ -10,6 +10,11 @@
 #include <tuple>
 #include <vector>
 
+#include <boost/version.hpp>
+#if BOOST_VERSION < 107400                      // TODO(anyone): remove this block once Ubuntu ships boost 1.74
+#include "utils/boost_bimap_core_override.hpp"  // NOLINT
```

Comment:
Again, the ancient boost version in Ubuntu becomes an issue. Even groovy (20.10) will come with 1.71. Use the same hack as we already successfully use with boost_curry_override.hpp.

Reply:
1. `#if BOOST_VERSION < 107400`  Call: `#include "utils/boost_bimap_core_override.hpp"`
2. Call: `#include <boost/bimap.hpp>` (l. 18)

Looks redundant to me. Shouldn't we put `#include <boost/bimap.hpp>` into an `#else`-clause?

Reply:
`bimap_core` is something different than `bimap`. `bimap.hpp` would usually include `bimap_core. By including it ourselves before `bimap.hpp` had a chance to do it, the backported version gets loaded and sets its header guards. As such, when `bimap.hpp` include the `bimap_core.hpp` that comes with the system's boost version, the header guard is already set, and the file will be ignored.

Fun fact: This hack would not be possible with `#pragma once`. Not sure if that is a good or a bad thing though ;)